src/array/cpu/id_hash_map.h

```diff
@@ -0,0 +1,141 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file array/cpu/id_hash_map.h
+ * @brief Class about id hash map
+ */
+
+#ifndef DGL_ARRAY_CPU_ID_HASH_MAP_H_
+#define DGL_ARRAY_CPU_ID_HASH_MAP_H_
+
+#include <dgl/aten/types.h>
+#include <dgl/runtime/device_api.h>
+
+#include <memory>
+#include <vector>
+
+namespace dgl {
+namespace aten {
+
+/**
+ * @brief A CPU targeted hashmap for mapping duplicate and non-consecutive ids
+ * in the provided array to unique and consecutive ones. It utilizes
+ * multi-threading to accelerate the insert and search speed. Currently it is
+ * only designed to be used in `ToBlockCpu` for optimizing.
+ *
+ * The hashmap should be used in two phases. With the first being creating the
+ * hashmap, and then init it with an id array. After that, searching any old ids
+ * to get the mappings according to your need.
```

Comment:
I don't quite understand this explanation.  Looks like it has three phases (creation, init, and search).

Do you want to say that querying before inserting all elements is not supported?  If so you can just say that in the limitation above.

Reply:
changed for remove misunderstanding.